src/app/frontend/common/services/global/history.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Injectable} from '@angular/core';
+import {RawParams, StateService, TransitionPromise, TransitionService} from '@uirouter/core';
+import {KdStateService} from './state';
+
+@Injectable()
+export class HistoryService {
+  private previousStateName: string;
+  private previousStateParams: RawParams;
+
+  constructor(private readonly kdState: KdStateService, private readonly state: StateService) {
+    this.init();
+  }
+
+  /** Initializes the service. **/
+  init(): void {
+    this.kdState.onSuccess.asObservable().subscribe(transition => {
```

Comment:
`asObservable` is not needed here, you can directly use `subscribe` method.

Reply:
Angular's EventEmitter extends subscribe and strips off the generic type info. The asObservable call is to bring the type back.

Reply:
Just for typing you can simply use `onSuccess.subscribe((transition: Transition) => {...});`.